### PR TITLE
Add configurable retry logic when Rancher responds with 405 method not allowed for node templates

### PR DESCRIPTION
**Pull request by rawmind0**
Related to https://github.com/rancher/rancher/issues/26456 and https://github.com/rancherlabs/eio/issues/126
Requires https://github.com/rancher/rancher/pull/27804

### CODE REVIEW

**Review by paraglade — APPROVED** [review]
LGTM

**Comment by superseb** [comment]
Logic looks fine, I think the retries documentation is now incorrect as its saying `Number of retries to connect to Rancher` and it's now being used as a generic retry setting which is used for retrying any connection that needs it.

**Comment by izaac** [comment]
Building the rancher2 provider from this fork/branch and validating #26456 results in no more `NotAllowed` errors for me.
https://github.com/rancher/rancher/issues/26456#issuecomment-659695546

File: rancher2/util.go [hunk 1]
```diff
@@ -266,6 +266,16 @@ func IsForbidden(err error) bool {
 	return apiError.StatusCode == http.StatusForbidden
 }
 
+// IsNotAllowed checks if the given APIError is a Forbidden HTTP statuscode
```
Comment (armsnyder) [inline rancher2/util.go]: This comment lies.

```suggestion
// IsNotAllowed checks if the given APIError is a Method Not Allowed HTTP statuscode
```

Comment (rawmind0) [reply rancher2/util.go]: More than a lie, it's a typo. Fixed

**Comment by rawmind0** [comment]
@superseb , updated retries documentation.

**Comment by superseb** [comment]
LGTM

**Pull request merged**